netlogo-gui/src/main/app/WidgetActions.scala

```diff
@@ -1,57 +1,104 @@
 package org.nlogo.app
 
 import java.awt.Rectangle
+import javax.swing.JComponent
+import javax.swing.undo.{AbstractUndoableEdit, UndoManager, UndoableEdit}
 
 import org.nlogo.core.Widget
+import org.nlogo.window.MouseMode
+
+class WidgetUndoManager extends UndoManager{
```

Comment:
I think there is no need to define this class (as you're overriding two methods by just calling the `super`'s counterparts, which I don't see the point of), you can just use UndoManager directly.